Tasks/CopyFilesToMachines/CopyFilesToMachines.ps1

```diff
@@ -37,68 +37,17 @@ function Output-ResponseLogs
     }
 }
 
-$CopyJob = {
-param (
-    [string]$environmentName,
-    [guid]$envOperationId,
-    [string]$fqdn, 
-    [string]$sourcePath,
-    [string]$targetPath,
-    [string]$username,
-	[string]$password,
-    [string]$cleanTargetBeforeCopy
-    )
-
-    Get-ChildItem $env:AGENT_HOMEDIRECTORY\Agent\Worker\*.dll | % {
-    [void][reflection.assembly]::LoadFrom( $_.FullName )
-    Write-Verbose "Loading .NET assembly:`t$($_.name)" -Verbose
-    }
-
-    Get-ChildItem $env:AGENT_HOMEDIRECTORY\Agent\Worker\Modules\Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs\*.dll | % {
-    [void][reflection.assembly]::LoadFrom( $_.FullName )
-    Write-Verbose "Loading .NET assembly:`t$($_.name)" -Verbose
-    }
-
-   $resOperationId = Invoke-ResourceOperation -EnvironmentName $environmentName -ResourceName $fqdn -EnvironmentOperationId $envOperationId -ErrorAction Stop
-
-   Write-Verbose "ResourceOperationId = $resOperationId for resource $fqdn" -Verbose
-
-   $credential = New-Object 'System.Net.NetworkCredential' -ArgumentList $username, $password
-
-   Write-Verbose "Initiating copy on $fqdn, username: $username" -Verbose
-
-   if($cleanTargetBeforeCopy -eq "true")
-    {
-         $copyResponse = Copy-FilesToTargetMachine -MachineDnsName $fqdn -SourcePath $sourcePath -DestinationPath $targetPath -Credential $credential -CleanTargetPath -SkipCACheck -UseHttp
-    }
-
-    else
-    {
-         $copyResponse = Copy-FilesToTargetMachine -MachineDnsName $fqdn -SourcePath $sourcePath -DestinationPath $targetPath -Credential $credential -SkipCACheck -UseHttp
-    }
-
-    $logs = New-Object 'System.Collections.Generic.List[System.Object]'
-    $log = "Deployment Logs : " + $copyResponse.DeploymentLog + "`nService Logs : " + $copyResponse.ServiceLog               
-    $resourceOperationLog = New-OperationLog -Content $log
-    $logs.Add($resourceOperationLog)
-
-	Complete-ResourceOperation -EnvironmentName $environmentName -EnvironmentOperationId $envOperationId -ResourceOperationId $resOperationId -Status $copyResponse.Status -ErrorMessage $copyResponse.Error -Logs $logs -ErrorAction Stop
-    
-    Write-Output $copyResponse
-}
-
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs"
 
-$env:DTL_ALTERNATE_CREDENTIALS_USERNAME = $alternateCredentialsUsername
-$env:DTL_ALTERNATE_CREDENTIALS_PASSWORD = $alternateCredentialsPassword
+$connection = Get-VssConnection -TaskContext $distributedTaskContext
 
-$resources = Get-EnvironmentResources -EnvironmentName $environmentName -ResourceFilter $machineNames -ErrorAction Stop
+$resources = Get-EnvironmentResources -EnvironmentName $environmentName -ResourceFilter $machineNames -ErrorAction Stop -Connection $connection
```

Comment:
Have the -EA Stop in the end for all the cmdlets call here and the parameters to the cmdlets itself should be together.